packages/ui/src/ui/pages/query-tracker/QueryFilesButton/index.tsx

```diff
@@ -1,39 +1,48 @@
 import React, {FC, useCallback, useEffect, useRef, useState} from 'react';
 import cn from 'bem-cn-lite';
 import {QueryFile} from '../module/api';
-import {Button, Icon, Text, Toaster} from '@gravity-ui/uikit';
+import {Button, Icon, Text} from '@gravity-ui/uikit';
 import clipIcon from '@gravity-ui/icons/svgs/paperclip.svg';
-import {VALIDATOR_ERRORS_TEXT, formValidator} from '../QuerySettingsButton/formValidator';
+import {VALIDATOR_ERRORS_TEXT} from '../QuerySettingsButton/formValidator';
 import './QueryFilesButton.scss';
 import {PopupWithCloseButton} from '../QuerySettingsButton/PopupWithCloseButton';
-import {FilesAddForm} from './FilesAddForm';
-import {
-    SaveFormData,
-    Props as SettingsItemEditFormProps,
-} from '../QuerySettingsButton/SettingsItemForm';
-import {setAccountLimit} from '../../../utils/accounts/editor';
+import {ActionsWithAddForm} from './ActionsWithAddForm';
 import {FilesTabs} from './FilesTabs';
+import {FileValidator, ValidatorError} from './FileItemForm';
+import {useDispatch, useSelector} from 'react-redux';
+import {
+    QueryFileAddForm,
+    QueryFileEditor,
+    selectAddForm,
+    selectDeletedFiles,
+    setAddForm,
+    setFileEditor,
+} from '../module/queryFilesForm/queryFilesFormSlice';
+import {changeQueryFile, deleteQueryFile, restoreQueryFile} from '../module/queryFilesForm/actions';
+import {getQueryFiles} from '../module/query/selectors';
+import {updateQueryDraft} from '../module/query/actions';
+import {createSelector} from 'reselect';
 
 const b = cn('query-files-popup');
-const toaster = new Toaster();
 
-enum FileTabs {
-    Current = 'current',
-    Deleted = 'deleted',
-}
-
-type Props = {
-    files: QueryFile[];
-    queryId: string;
-    onChange: (files: QueryFile[]) => void;
-};
-
-export const QueryFilesButton: FC<Props> = ({files, onChange}) => {
+const selector = createSelector(
```

Comment:
It is better to follow common approaches and move the selector to a separate file near the definition of the corresponding reducer.